### DISCUSSION TITLE
What could cause "Storage Move lockup"?

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#150] [General]
Something I noted in the direct file coding under user library in nos 2.

Some kind of race condition?

**Comment by kej715** [comment]
That section is discussing, in particular, rules about resource reservation
by PP programs. When a PP program is assigned to a control point, the
control point cannot be moved. Therefore, PP programs must be careful to
avoid deadlocks related to reserving resources such as channels,
particularly when they need to reserve more than one reserve at a time. If
a were to become deadlocked, the control point it is serving cannot be
moved, and this could prevent control points around it from acquiring
additional memory. Hence, "Store Move Lockup" would occur.

**Reply by KC5VDJ** [reply]
Ah, thanks.  That makes sense.
